src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProject.cs

```diff
@@ -291,6 +291,18 @@ private static IEnumerable<PackageReference> GetPackageReferences(IEnumerable<Li
             return true;
         }
 
+        public override Task<string> GetCacheFilePathAsync()
+        {
+            var spec = GetPackageSpec();
+            if (spec == null)
+            {
+                throw new InvalidOperationException(
+                    string.Format(Strings.ProjectNotLoaded_RestoreFailed, ProjectName));
+            }
+
+            return Task.FromResult(NoOpRestoreUtilities.GetProjectCacheFilePath(cacheRoot: spec.RestoreMetadata.OutputPath, projectPath: spec.RestoreMetadata.ProjectPath));
```

Comment:
@dtivel The code has evolved to the point where RestoreMetadata is never null. 
The other null check there was just something that was overseen, and is hit by all transitive restore codepaths. 
The spec value here is the one created during nomination, so it has to go through https://github.com/NuGet/NuGet.Client/blob/2664e37a4cff5f9e5313088ca29741ebd81502ef/src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs#L222-L289 for project restore and https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/ToolRestoreUtility.cs#L22-L58 for tool restore.